### PR TITLE
Added small block bonus to block_while_worn items

### PR DESCRIPTION
**Pull request by Hymore246**
#### Summary

```SUMMARY: Balance "Added small block bonus to block_while_worn items"```

#### Purpose of change

All items with a BLOCK_WHILE_WORN tag have a block bonus of 2 if they have no block technique,

#### Describe the solution

Arm guards and other items that can be used to block attacks while worn didn't provide any bonus to blocking attacks. Without going into heavy detail by breaking down the block bonus formula from the Block_Hit() method in melee.cpp, I can say that a BLOCK_WHILE_WORN item without a block technique is worse at blocking than a player with arm/leg block and Unarmed 1. 

To fix this, I made it so BLOCK_WHILE_WORN items give a block bonus of 2 if they do not have a block technique. This bonus only applies while the item is being worn. For comparison, items with WBLOCK_1 have a block bonus of 4.

In addition, I added a BLOCK_WHILE_WORN tag to the steel arm guards which should have had the tag to begin with.

### CODE REVIEW

**Comment by RarkGrames** [comment]
Isn't WBLOCK_1 only applies to wielded items? So to use WBLOCK_1, you have to wield the arm guards instead of wearing them. I thought that 2-by-arm guards having this is a mistake.

**Comment by Hymore246** [comment]
It would apply if you wield them, but also applies if worn with the BLOCK_WHILE_WORN tag.

```
static int blocking_ability( const item &shield )
{
    int block_bonus = 0;
    if( shield.has_technique( WBLOCK_3 ) ) {
        block_bonus = 10;
    } else if( shield.has_technique( WBLOCK_2 ) ) {
        block_bonus = 6;
    } else if( shield.has_technique( WBLOCK_1 ) ) {
        block_bonus = 4;
    }
    return block_bonus;
}

item &player::best_shield()
{
    // Note: wielded weapon, not one used for attacks
    int best_value = blocking_ability( weapon );
    item *best = best_value > 0 ? &weapon : &null_item_reference();
    for( item &shield : worn ) {
        if( shield.has_flag( "BLOCK_WHILE_WORN" ) && blocking_ability( shield ) >= best_value ) {
            best = &shield;
        }
    }

    return *best;
}
```

**Comment by Hymore246** [comment]
I talked to ChaosVolt and in an effort to avoid making shields even worse, I've changed the PR to giving a block bonus of 2 to BLOCK_WHILE_WORN items instead of giving them WBLOCK_1. Without any sort of block bonus, the game will ignore blocking with items and will try to use limb blocking instead.

File: src/melee.cpp [hunk 1]
```diff
@@ -1315,6 +1315,8 @@ static int blocking_ability( const item &shield )
         block_bonus = 6;
     } else if( shield.has_technique( WBLOCK_1 ) ) {
         block_bonus = 4;
+    } else if ( shield.has_flag( "BLOCK_WHILE_WORN" ) ) {
```
Comment (kevingranade) [inline src/melee.cpp]:
```suggestion
    } else if( shield.has_flag( "BLOCK_WHILE_WORN" ) ) {
```

**Pull request merged**